### ISSUE TITLE
`require.context()` is not transformed in `.mjs` files

### ISSUE THREAD

**Issue by sirlancelot** [webpack/webpack#6690]
**Do you want to request a *feature* or report a *bug*?** Bug

**What is the current behavior?** `require.context()` is not transformed

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
// entry.mjs

const pages = require.context("./pages/", true, /^\.\/.*\.vue$/, "lazy")
const routes = pages.keys().map((name) => ({
    component: () => pages(name),
    path: "/" + name.slice(2, -4)
}))
```

**What is the expected behavior?** `require.context()` is transformed in to a `webpackAsyncContext` module (consistent with the behavior as if this statement were in a `.js` file).

**If this is a feature request, what is motivation or use case for changing the behavior?** N/A

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

`node v8.9.3`
`webpack v4.1.0`
`@std/esm v0.24.0`

I'm running Webpack like so: `node -r @std/esm ./node_modules/.bin/webpack [options]` so that I can write the config files using ESModules.

**Comment by jdalton** [comment]
Hi @sirlancelot!

Webpack 4 supports the `-r` flag so should be able to do `webpack -r @std/esm [options]`.

**Comment by sirlancelot** [comment]
@jdalton nice, I wasn't aware of that!

The main issue still exists though.

**Comment by ooflorent** [comment]
`require` and `module` is not supported for MJS files. That's the expected behavior.

**Comment by ooflorent** [comment]
### tl;dr

`require` and `module` is not supported for MJS files. That's the expected behavior.

### details

In webpack 4, when a filename ends with `.mjs` it is parsed as a strict Harmony module (internally called `javascript/esm`). In webpack 3, this mode didn't exist and a hybrid mode was used (called `javascript/auto` in v4). These 2 modes behave differently:

- When `javascript/auto` is used, the file is parsed as a _module_ but it can use CJS bindings. This mode permits mixing `require` calls and `import` statements. 

- When `javascript/esm` is used, the file is also parsed as a _module_ but CJS bindings are not injected. `require` and `module` are not available and there are transformed.

**Comment by jdalton** [comment]
@ooflorent It would be nice if webpack informed the user of this when failing to transform `require.context`.

@sirlancelot The good news is you can configure `@std/esm` and webpack to use `.js`.

For `std/esm` you can use the `"@std/esm":"cjs"` option and for webpack 4 you can use `module.rules[].type` of `"javascript/auto"`.

**Comment by ooflorent** [comment]
@jdalton I've created an issue (#6698) to address this.
